pkg/pillar/devicenetwork/dnc.go

```diff
@@ -251,7 +251,13 @@ func VerifyPending(pending *DPCPending,
 	pending.PendDNS = pend2
 
 	// We want connectivity to zedcloud via atleast one Management port.
-	rtf, err := VerifyDeviceNetworkStatus(pending.PendDNS, 1, timeout)
+	rtf, err := VerifyDeviceNetworkStatus(&pending.PendDNS, 1, timeout)
+	// Copy errors from PendDns to PendDpc
+	for indx := range pending.PendDNS.Ports {
+		pending.PendDPC.Ports[indx].Error = pending.PendDNS.Ports[indx].Error
+		pending.PendDPC.Ports[indx].ErrorTime =
+			pending.PendDNS.Ports[indx].ErrorTime
+	}
```

Comment:
This seems really clunky and brittle; can't even tell whether or not there is missing code to clear the errors.
I think it would be cleaner to have VerifyDeviceNetworkStatus not modify PendDNS but instead return a struct with the per interface errors seen by that particular verification. Then the caller can decide to update based on that is in that error struct, and there is no lingering errors in the DeviceNetworkStatus; the per interface errors are only reported as part of the DPC. Such code I can at least have a chance to verify that the errors are cleared up by subsequent attempts.